src/main/java/eu/tsystems/mms/tic/testerra/plugins/xray/jql/predefined/IssueType.java

```diff
@@ -27,23 +27,29 @@
 import eu.tsystems.mms.tic.testframework.common.PropertyManager;
 
 public enum IssueType {
-    Test(new JiraIssueType(PropertyManager.getProperty(XrayMapper.PROPERTY_TEST_ISSUETYPE_NAME, "Test"))),
-    TestExecution(new JiraIssueType(PropertyManager.getProperty(XrayMapper.PROPERTY_TEST_EXECUTION_ISSUETYPE_NAME, "Test Execution"))),
-    TestSet(new JiraIssueType(PropertyManager.getProperty(XrayMapper.PROPERTY_TEST_SET_ISSUETYPE_NAME,"Test Set"))),
+    Test,
+    TestExecution,
+    TestSet,
     ;
 
-    private final JiraIssueType issueType;
-
-    IssueType(JiraIssueType literal) {
-        this.issueType = literal;
+    IssueType() {
     }
 
     @Override
     public String toString() {
-        return issueType.getName();
+        return getIssueType().getName();
     }
 
+    // Properties can be set later after class loading, so resolve issue type at runtime
     public JiraIssueType getIssueType() {
-        return this.issueType;
+        switch ( this ) {
```

Comment:
FWIW: You can get around the `default` and its assorted maintenance chore by making each enum value its own anonymous subclass of `IssueType`. Using an interface declaring `getIssueType` implemented by `IssueType` each new value will cause a compile time error if it does not implement `getIssueType` => no chance to forget an implementation and actually do throw.

... Admittedly that does look a little ugly, but at least not at runtime.

Reply:
OK, Commited as e1439d2d66a7d477cd2d